### ISSUE TITLE
Pausing and Resuming Model Training in YOLO

### ISSUE THREAD

**Issue by AISoltani** [ultralytics/ultralytics#19466]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi everyone,
Thank you for all the amazing work you do! I know Glenn Jocher will likely answer this, as he’s always been a pioneer in providing solutions. :)

My question is: **Is it possible to pause and resume the training process in YOLO?**

Here’s the scenario:

- I have a training process with 300 epochs.

- Due to limited GPU resources (e.g., I’m using Kaggle, which provides 30-hours free GPU per week also 12 hour per each session), I need to pause training at, say, epoch 40, and then resume it later.

I’m aware that I can save the best model checkpoint during training and use it as a base for a new training session. However, the issue is that when I use the best model as a base, the scores often drop significantly in the new training session, and it takes a long time to recover. This is especially problematic when my dataset is complex, then each epoch takes about 1 hour to train.

By the time the model recovers from the drop, I’ve already spent a lot of time and resources, which feels inefficient.

I hope I’ve explained my challenge clearly. If there’s a solution or a better approach to handle this, I’d greatly appreciate your advice!

Thank you in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AISoltani, thank you for your interest in Ultralytics 🚀! We appreciate you taking the time to search for existing issues and providing a clear description of your question.

To address your question about pausing and resuming training, we recommend exploring the functionality provided by **resume training** within the Ultralytics framework, which builds upon saved checkpoints. For detailed insights, please refer to the [Docs](https://docs.ultralytics.com/) where many training workflows, including resuming from checkpoints, are documented. You can find practical examples in [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) sections.

If this persists as a concern or potential bug 🐛, we encourage you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/), including specifics such as your training configuration, hardware setup, and detailed logs. This will help us investigate further and provide a tailored solution.

Alternatively, you can also join the Ultralytics community to discuss various tips on training efficiency and resources:

- 🎧 [Discord](https://discord.com/invite/ultralytics) for real-time chat support
- 💬 [Discourse](https://community.ultralytics.com/) for in-depth discussions
- 📚 [Subreddit](https://reddit.com/r/Ultralytics) for knowledge sharing and ideas 

### Upgrade
Ensure that you are running the latest `ultralytics` package in a compatible environment:

```bash
pip install -U ultralytics
```

Check our [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) file for dependencies and confirm your setup uses [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

### Environments
You can use free or cloud-based platforms to train your models in sessions aligned with your GPU time constraints. These verified environments should help make your workflow smoother:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

- **Google Cloud** Deep Learning VM: See our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)

- **Amazon** Deep Learning AMI: Refer to our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)

- **Docker Image**: Learn more in the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

⚡ This is an automated response to help address your issue promptly. An Ultralytics engineer will review your question and get back to you soon! 😊

**Comment by Y-T-G** [comment]
You can load the last.pt and use `model.train(resume=True)` to resume. You can only resume if the training was interrupted and didn't complete the specified epochs.

**Comment by AISoltani** [comment]
> You can load the last.pt and use `model.train(resume=True)` to resume. You can only resume if the training was interrupted and didn't complete the specified epochs.

Dear @Y-T-G 
That;s Great, Thanks for your reply!

**Comment by Y-T-G** [comment]
`last.pt` always saved after epoch. `save_period` is if you want to save checkpoints without overwriting old ones
